data/transactions/logic/doc.go

```diff
@@ -212,45 +218,48 @@ func OpImmediateNote(opName string) string {
-	"app_local_get":     "params: Txn.Accounts offset (or, since v4, an account address that appears in Txn.Accounts or is Txn.Sender), state key. Return: value. The value is zero (of type uint64) if the key does not exist.",
-	"app_local_get_ex":  "params: Txn.Accounts offset (or, since v4, an account address that appears in Txn.Accounts or is Txn.Sender), application id (or, since v4, a Txn.ForeignApps offset), state key. Return: did_exist flag (top of the stack, 1 if exist and 0 otherwise), value. The value is zero (of type uint64) if the key does not exist.",
-	"app_global_get_ex": "params: Txn.ForeignApps offset (or, since v4, an application id that appears in Txn.ForeignApps or is the CurrentApplicationID), state key. Return: did_exist flag (top of the stack, 1 if exist and 0 otherwise), value. The value is zero (of type uint64) if the key does not exist.",
-	"app_global_get":    "params: state key. Return: value. The value is zero (of type uint64) if the key does not exist.",
-	"app_local_put":     "params: Txn.Accounts offset (or, since v4, an account address that appears in Txn.Accounts or is Txn.Sender), state key, value.",
-	"app_local_del":     "params: Txn.Accounts offset (or, since v4, an account address that appears in Txn.Accounts or is Txn.Sender), state key.\n\nDeleting a key which is already absent has no effect on the application local state. (In particular, it does _not_ cause the program to fail.)",
-	"app_global_del":    "params: state key.\n\nDeleting a key which is already absent has no effect on the application global state. (In particular, it does _not_ cause the program to fail.)",
-	"asset_holding_get": "params: Txn.Accounts offset (or, since v4, an account address that appears in Txn.Accounts or is Txn.Sender), asset id (or, since v4, a Txn.ForeignAssets offset). Return: did_exist flag (1 if exist and 0 otherwise), value.",
-	"asset_params_get":  "params: Before v4, Txn.ForeignAssets offset. Since v4, Txn.ForeignAssets offset or an asset id that appears in Txn.ForeignAssets. Return: did_exist flag (1 if exist and 0 otherwise), value.",
-	"app_params_get":    "params: Txn.ForeignApps offset or an app id that appears in Txn.ForeignApps. Return: did_exist flag (1 if exist and 0 otherwise), value.",
-	"log":               "`log` can be called up to MaxLogCalls times in a program, and log up to a total of 1k bytes.",
+	"ed25519verify":       "The 32 byte public key is the last element on the stack, preceded by the 64 byte signature at the second-to-last element on the stack, preceded by the data which was signed at the third-to-last element on the stack.",
+	"ecdsa_verify":        "The 33 byte X-component of a public key is the last element on the stack, preceded by Y-component of a pubkey (can be empty, so the X-component then interpreted as a compressed pubkey), preceded by R and S components of a signature, preceded by the data that is fifth element on the stack.",
```

Comment:
It's best not to mention byte lengths of the public keys and signatures, since different curves may have different sizes for these.

Also it is worth mentioning that the signed data [must be exactly 32 bytes](https://github.com/ethereum/go-ethereum/blob/90987db7334c1d10eb866ca550efedb66dea8a20/crypto/secp256k1/secp256.go#L127) (really a hash of the message). It may also be worth mentioning that [only signatures "in lower-S form"](https://github.com/ethereum/go-ethereum/blob/d8ff53dfb8a516f47db37dbc7fd7ad18a1e8a125/crypto/secp256k1/libsecp256k1/include/secp256k1.h#L366) will be accepted.

Reply:
added